src/conf-yaml-loader.c

```diff
@@ -195,15 +195,15 @@ ConfYamlParse(yaml_parser_t *parser, ConfNode *parent, int inseq)
             yaml_version_directive_t *ver =
                 event.data.document_start.version_directive;
             if (ver == NULL) {
-                fprintf(stderr, "ERROR: Invalid configuration file.\n\n");
-                fprintf(stderr, "The configuration file must begin with the following two lines:\n\n");
-                fprintf(stderr, "%%YAML 1.1\n---\n\n");
+                SCLogError(SC_ERR_CONF_YAML_ERROR, "ERROR: Invalid configuration file.");
```

Comment:
I don't mind this change but I'm wondering why the change was restricted to this file? There are plenty of `fprintf` and `printf` calls elsewhere.

Reply:
These seemed the only ones reachable from fuzzing targets.

Doing `grep fprintf src/*.c | grep stderr`, I get 
```
src/conf-yaml-loader.c:                fprintf(stderr, "ERROR: Invalid configuration file.\n\n");
src/conf-yaml-loader.c:                fprintf(stderr, "The configuration file must begin with the following two lines:\n\n");
src/conf-yaml-loader.c:                fprintf(stderr, "%%YAML 1.1\n---\n\n");
src/conf-yaml-loader.c:                fprintf(stderr, "ERROR: Invalid YAML version.  Must be 1.1\n");
src/conf-yaml-loader.c:        fprintf(stderr, "Failed to initialize yaml parser.\n");
src/conf-yaml-loader.c:        fprintf(stderr, "final value was reassigned\n");
src/suricata.c:                    fprintf(stderr, "ERROR: Failed to set pfring.cluster-id.\n");
src/suricata.c:                    fprintf(stderr, "ERROR: Failed to set pfring.cluster-type.\n");
src/suricata.c:                    fprintf(stderr, "ERROR: Failed to set engine init-failure-fatal.\n");
src/suricata.c:                            fprintf(stderr, "ERROR: Failed to set unix-command.filename.\n");
src/suricata.c:                fprintf(stderr, "ERROR: Unit tests not enabled. Make sure to pass --enable-unittests to configure when building.\n");
src/suricata.c:                fprintf(stderr, "ERROR: Unit tests not enabled. Make sure to pass --enable-unittests to configure when building.\n");
src/suricata.c:                    fprintf(stderr, "ERROR: Failed to set erf-file.file\n");
src/suricata.c:                    fprintf(stderr, "ERROR: Failed to set pcap-buffer-size.\n");
src/suricata.c:                        fprintf(stderr, "Failed to set configuration value %s.",
src/suricata.c:                fprintf(stderr, "ERROR: Failed to set engine init-failure-fatal.\n");
src/suricata.c:            fprintf(stderr, "ERROR: Unit tests not enabled. Make sure to pass --enable-unittests to configure when building.\n");
src/util-byte.c:        fprintf(stderr, "ByteExtractString: Extra characters following numeric value\n");
src/util-byte.c:        fprintf(stderr, "ByteExtractStringSigned: Extra characters following numeric value\n");
src/util-byte.c:    //fprintf(stderr, "ByteExtractStringSigned: Extracted base %d: 0x%" PRIx64 "\n", base, *res);
src/util-unittest.c:                        fprintf(stderr, "ERROR: unittest failed.\n");
```

Maybe we can change the ones in src/util-byte.c
For the other ones, I am not sure logging is initialized before we call them (especially the ones in suricata.c)